File: apps/api/app/Models/Partner.php
<?php

namespace App\Models;

use Database\Factories\PartnerFactory;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Partner extends Model
{
    /** @use HasFactory<PartnerFactory> */
    use HasFactory;

    public const TYPE_FARMER = 'farmer';

    public const TYPE_SUPPLIER = 'supplier';

    protected $fillable = [
        'type',
        'name',
        'phone',
        'email',
        'address',
        'district',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
        ];
    }
}

File: apps/mobile/lib/features/partners/data/models/partner.dart
class Partner {
  const Partner({
    required this.id,
    required this.type,
    required this.name,
    required this.isActive,
    this.phone,
    this.email,
    this.address,
    this.district,
  });

  final int id;
  final String type;
  final String name;
  final String? phone;
  final String? email;
  final String? address;
  final String? district;
  final bool isActive;

  factory Partner.fromJson(Map<String, dynamic> json) {
    return Partner(
      id: json['id'] as int,
      type: json['type'] as String,
      name: json['name'] as String,
      phone: json['phone'] as String?,
      email: json['email'] as String?,
      address: json['address'] as String?,
      district: json['district'] as String?,
      isActive: json['is_active'] as bool? ?? true,
    );
  }
}
